src/OpenLoco/S5/S5.cpp

```diff
@@ -362,6 +374,312 @@ namespace OpenLoco::S5
         }
     }
 
+    // 0x00445A4A
+    static void fixState(GameState& state)
+    {
+        if (state.fixFlags & S5FixFlags::fixFlag0)
+        {
+            state.fixFlags |= S5FixFlags::fixFlag1;
+        }
+        if (!(state.fixFlags & S5FixFlags::fixFlag1))
+        {
+            // Shift data after companies to correct location
+            auto src = reinterpret_cast<uint8_t*>(&state) + 0x49EA24;
+            auto dst = src + 0x1C20;
+            for (size_t i = 0; i < 0x40E200; i++)
+            {
+                *--dst = *--src;
+            }
+
+            // Convert each company format from old to new
+            for (size_t i = 0; i < std::size(state.companies); i++)
+            {
+                for (size_t j = 0; j < 372; j++)
```

Comment:
Also std::memcpy?

Reply:
it's a bit annoying to do that as these are going in reverse.